packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml

```diff
@@ -500,6 +510,31 @@ spec:
         matchResources:
           controlPlane: true
       jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/initConfiguration/nodeRegistration/kubeletExtraArgs/node-labels
+        valueFrom:
+          template: |
+            {{ $first := true }}
+            {{- range $key, $val := (index .TKR_DATA .builtin.controlPlane.version).labels }}
+            {{- if $first }}
+              {{- $first = false }}
+            {{- else -}}
+              ,
+            {{- end }}
+            {{- $key }}={{ $val }}
+            {{- end }}
+            {{- if .controlPlane.nodeLabels -}}
+              ,
```

Comment:
If TKR_DATA.<VERSION>.labels is empty, the first comma should be removed.

Reply:
Yes, but there're always default labels in control plane, right? like os, arch, etc. If this (not likely) is a problem, maybe another PR to fix all the related part in the manifest.

Reply:
This is an implicit dependency. Can we add a comma only `if (index .TKR_DATA .builtin.controlPlane.version).labels`?

Reply:
Added on all Iaas